docs/release_guide.md

```diff
@@ -38,22 +38,23 @@ Identify the base commit in the `main` branch for the release and cut a release
 $ git checkout -b release-<version> <commit-id> # ex: git checkout -b release-v0.4 0d05587
 ```
 
-Push the release branch to the main repo (not fork), identified here by the `upstream` remote.
+Push the release branch to the upstream repo (not forked), identified here by the `upstream` remote.
 ```console
 $ git push upstream release-<version> # ex: git push upstream release-v0.4
 ```
 
+_Note_: Do not create a new branch for RCs in the upstream repo.
+
 ## Update release branch with patches and versioning changes
 
-Create a new branch off of the release branch to maintain updates specific to the new version.
+Create a new branch off of the release branch to maintain updates specific to the new version. Let's call it the patch branch. The patch branch should not be created in the upstream repo.
```

Comment:
> The patch branch should not be created in the upstream repo.

If we delete the branch after its PR is merged, I don't see why it makes a difference where the branch lives. I understand every branch is currently captured by the upstream repo's branch protection rules which means an admin has to delete the branch and that's cumbersome.

I think a better solution to that particular problem is to fix in the branch protection rules vs. qualifying every current and future instruction about pushing short-lived branches with this same statement. Keeping this seems fine to me until then.